lib/xcodeproj/project.rb

```diff
@@ -80,6 +80,7 @@ def initialize(path, skip_initialization = false, object_version = Constants::DE
       unless skip_initialization
         initialize_from_scratch
         @object_version = object_version.to_s
+        root_object.compatibility_version = Constants::COMPATIBILITY_VERSION_BY_OBJECT_VERSION.fetch(object_version)
```

Comment:
we use `fetch` here because if a new object_version is passed we want to ensure that the `COMPATIBILITY_VERSION_BY_OBJECT_VERSION` hash is updated.

Reply:
should explicitly test what happens when the object version is unknown, and ensure that a helpful error message is raised (with instructions to file an issue)

Reply:
done